webapp/src/js/enketo/widgets/db-object-widget.js

```diff
@@ -30,26 +30,38 @@ function construct(element) {
 
   const Select2Search = window.CHTCore.Select2Search;
 
-  let $textInput = $question.find('input');
-
-  const value = $textInput.val();
-  const disabled = $textInput.prop('readonly');
-  $textInput.replaceWith($textInput[0].outerHTML.replace(/^<input /, '<select ').replace(/<\/input>/, '</select>'));
-  $textInput = $question.find('select');
-  const preSelectedOption = $('<option></option>')
-    .attr('value', value)
-    .text(value);
-  $textInput.append(preSelectedOption);
-
-  const contactTypes = getContactTypes($question, $textInput);
+  const $textInput = $question.find('input');
+  const $proxyInput = $textInput.clone();
+
+  const $option = $('<option></option>');
+  const setOptionValue = value => $option.attr('value', value).text(value);
+  setOptionValue($textInput.val());
+  $textInput.on('inputupdate', () => setOptionValue($textInput.val()));
```

Comment:
For all intents and purposes, I like this refactor. 
Turning an `input` into a `select` just makes me nervous ...

Reply:
I'm not meaning to add more work on your plate but, since we're already refactoring this, what is the value of cloning the input and then changing the tag versus just making a new select element? 
Is it because there are classes and attributes to copy? Are they actually needed on the select? 
This is similar to my question above about the duplicated name. 

Could there be bugs if we suddenly have two elements that match an old query, be it an attribute or name search?

Reply:
Good call on the cloning!  I had forgotten to circle back to that after I got everything else cleaned up. I went back and did some more testing and found that I could just use a brand new `select` element without copying anything (besides the `name` as discussed above).  

I think the only thing we need to decide here is what we should use as the name for the the `select`s.  I have left the name of the `select` the same as the `input` for now since it simplifies logic downstream in the `changeHandler`, but we can change that (as I mentioned above). I would just need to include some other way of getting the field path into the `changeHandler` function...